portality/upgrade.py

```diff
@@ -100,7 +102,16 @@ def do_upgrade(definition, verbose):
                 # When we have enough, do some writing
                 if len(batch) >= batch_size:
                     total += len(batch)
+                    batch_num += 1
+
                     print(datetime.now(), "writing ", len(batch), "to", tdef.get("type"), ";", total, "of", max)
+
+                    if save_batches:
+                        fn = os.path.join(save_batches, tdef.get("type") + "." + str(batch_num) + ".json")
+                        with open(fn, "w") as f:
+                            f.write(json.dumps(batch, indent=2))
+                            print(datetime.now(), "wrote batch to file {x}".format(x=fn))
```

Comment:
those lines of code for save batches is very common.
consider to extract as a function